lms/services/lti_grading/factory.py

```diff
@@ -1,27 +1,48 @@
 from lms.services.lti_grading._v11 import LTI11GradingService
 from lms.services.lti_grading._v13 import LTI13GradingService
+from lms.services.lti_grading.interface import LTIGradingService
 from lms.services.ltia_http import LTIAHTTPService
 
 
-def service_factory(_context, request):
-    application_instance = request.lti_user.application_instance
+def service_factory(_context, request, application_instance=None) -> LTIGradingService:
```

Comment:
The factory needed this extra parameter.

I reckon is a good compromise that allows us to use the same factory but not the find_request mechanism.